dhis-2/dhis-services/dhis-service-core/src/test/java/org/hisp/dhis/trackedentity/TrackedEntityAttributeStoreTest.java

```diff
@@ -144,6 +168,64 @@ public void testSaveTrackedEntityAttribute()
         assertNotNull( attributeService.getTrackedEntityAttribute( idB ) );
     }
 
+    @Test
+    public void testGetAllIndexableAttributes()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        long idB = attributeService.addTrackedEntityAttribute( attributeY );
+
+        Set<TrackedEntityAttribute> indexableAttributes = attributeService
+            .getAllTrigramIndexableTrackedEntityAttributes();
+
+        assertNotNull( indexableAttributes );
+        assertEquals( indexableAttributes.size(), 9 );
+        assertTrue( indexableAttributes.contains( attributeW ) );
+        assertTrue( indexableAttributes.contains( attributeY ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'A' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'E' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'J' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'N' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'O' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'S' ) ) );
+        assertTrue(
+            indexableAttributes.contains( attributeService.getTrackedEntityAttributeByName( "Attribute" + 'T' ) ) );
+    }
+
+    @Test
+    public void testCreateTrigramIndex()
+    {
+        long idA = attributeService.addTrackedEntityAttribute( attributeW );
+        Exception exception = assertThrows( BadSqlGrammarException.class,
+            () -> attributeService.createTrigramIndex( attributeW ) );
+        assertTrue( exception.getMessage().contains( String.format(
+            "CREATE INDEX CONCURRENTLY IF NOT EXISTS in_gin_teavalue_%d ON trackedentityattributevalue USING gin (trackedentityinstanceid,lower(value) gin_trgm_ops) where trackedentityattributeid = %d",
+            idA, idA ) ) );
+    }
+
+    @Test
+    public void testRunAnalyze()
+    {
+        Exception exception = assertThrows( BadSqlGrammarException.class,
+            () -> attributeService.runAnalyze() );
+        assertTrue( exception.getMessage().contains(
+            "ANALYZE trackedentityattributevalue" ) );
```

Comment:
I guess this is hard to test, but this doesn't feel too good to assert on the SQL statement and also to somehow provoke exceptions to do so 🤔

Reply:
Same as above. I think I can do it by mocking and testing the mock is called correctly. Will try that!

Reply:
changed this test.